### DISCUSSION TITLE
Is there a list of recommended pumps compatible with NPC?

### DISCUSSION THREAD

**Discussion by psmith3** [tagyoureit/nodejs-poolController#523] [Nixie]
Time to buy a new pump and I am looking to go to a variable speed and not spend a gazillion dollars. I currently have a 1.65hp single speed. I found various posts of different pumps that people are using, but is there a combined list? If not, what would be the best recommendation? Hayward VS?

**Comment by rstrouse** [comment]
The supported pumps are as follows in the order which I like them for automation and capabilities.

Pentair IntelliFlo VSF
Pentair IntelliFlo VS
Sta-Rite IntelliPro VSF
Pentair IntelliFlo XF  (If you are not running 3" suction, the additional cost of the XF pump series is probably not worth it) Avoid the 2-speed models
Sta-Rite IntelliPro XF

While Nixie will control the Hayward pumps I am not as familiar with the models.  However, there are a few folks with them here that seem happy with them.  The one drawback is that their underlying drive does not seem all that accurate with RPM and Energy use but it may not be all that important.
Hayward TriStar
Hayward EcoStar

While Nixie supports these pumps you should avoid them if you do not need a Safety Vacuum Relief system.  They shut down if there is restriction and you will see tons of threads on TFP with people trying to get their pumps to not false detect entrapment.
Pentair IntelliFlo VS+SVRS
Pentair IntelliPro VS+SVRS

Pentair also offers the IntelliFlo i1 and i2 series.  The i1 is for restricted flow systems but it you have a 1.65hp pump already I would shy away.  The i2 is for systems that can only accommodate a 15A breaker on the panel.  If it were me I would be more inclined to get the high performance versions (non i1 & i2) since they have some of the best low rpm flow ratings which equates to energy savings.

Although I have not seen them there is also supposed to be a WhisperFlo VS and SuperFlo VS line that has RS485 compatability.  Supposedly they use the same protocol as the IntelliFlo VS but require the automation cable to be purchased separately.  This adds about $70 to the cost of the pump.

-- The following relay driven variable speed pumps are also supported.  These do not report rpm, gpm, or energy use.
Hayward VS (non RS485)
Pentair Superflo VST (non RS485)

All that being said the Pentair IntelliFlo VSF would get my vote followed by the IntelliFlo VS.  It is a 3hp pump but has some of the best numbers in terms of energy use, versatility, and long term performance.  These pumps are identical in terms of performance but the VSF will report flow as well as RPM and energy use.

Bear in mind this is just my opinion based upon the information I hear from folks out there using, automating, and buying them.  Then again I remember a time when you could get an IntelliFlo VSF for < $1k and have it delivered in 3 days.

**Reply by tagyoureit** [reply]
And not on the list are the VF only pumps.  Everyone seems to want the flow rate so they can calculate turns easily but they are wildly inaccurate and based more on the motor speed and resistance than actual flow.  The RPMS are always more accurate.

**Reply by mguinness** [reply]
Just a nit, the WhisperFlo and SuperFlo VS line don't have RS485 compatibility, but the VST line does.  Interesting the manuals for the VST state they aren't compatible with the IntelliConnect system, but are with the EasyTouch/IntelliCenter - different protocol/command set?

EDIT:  I've since learnt that the T after VS signifies [TradeGrade](https://www.pentair.com/en-us/education-support/water-education-center/pool-and-spa-education/pentair-tradegrade-consumer.html) so apologies for any confusion.

**Reply by rstrouse** [reply]
I don't really know about the IntelliConnect protocol but I do suspect that it is probably an old cut down version of the IntelliTouch code.  SunTouch is kind of the same.  

It is possible that the timing on the new drives needs to be faster than the IntelliConnect can put out to retain control of the drive.  The other thing that IntelliConnect might do is use the old pump schedule commands on the drive but to be sure one would need to see a packet capture when IntelliConnect is communicating with the pump.  This would say it all.

**Reply by mguinness** [reply]
The IntelliConnect can communicate with the Intelliflo so the timing seems unlikely.  The older EasyTouch works with these new VST pumps, so I don't know if they would use the new pump schedule commands.  As you say a packet capture would tell the story.  The Superflo VST was manufactured from Oct 2020 and the WhisperFlo VST from May 2022 so hopefully someone will be able to do that in the near future.

**Comment by psmith3** [comment]
Perfect Reply! Thanks so much for all this information. I hate to throw questions like this out there when I see just how many emails and posts that you respond too! Great information and much appreciated. I follow your entire github account for issues and responses and see all of the communications! It is a lot.

On Jul 11, 2022, at 7:09 PM, tagyoureit ***@***.***> wrote:

﻿

And not on the list are the VF only pumps. Everyone seems to want the flow rate so they can calculate turns easily but they are wildly inaccurate and based more on the motor speed and resistance than actual flow. The RPMS are always more accurate.

**Comment by mguinness** [comment]
It looks like someone was able to control the Superflo VST pump with Nixie as detailed in https://github.com/tagyoureit/nodejs-poolController/issues/457.

A [packet log](https://github.com/tagyoureit/nodejs-poolController/files/9378179/packetLog.log) from that issue.  That means the protocol and commands are the same as the Intelliflo?  

Would it be worthwhile create a new entry for this pump (and most likely the Whisperflo VST as well)?

https://github.com/tagyoureit/nodejs-poolController/blob/46c3047e950f69b3ae4cf581ffdae64205271d5f/controller/boards/NixieBoard.ts#L77-L86

**Reply by rstrouse** [reply]
The fix in there was to remove the program set function since VST pumps don't have this feature.  We weren't using it anyway and IntelliCenter no longer emits it on the VS protocol.  So perhaps this is the difference in that any comms failure with IntelliConnect results in a pump communication failure.  So the action 5 causes the rest of the communication chain to stop.

If we did add the VSTs they would map internally to the vs pumps since the protocols are now identical.

**Reply by mguinness** [reply]
I'm not aware that anyone has tried to use it with the IntelliConnect as in the [Installation and Users Guide](https://www.pentair.com/content/dam/extranet/nam/pentair-pool/residential/manuals/pool-pumps/superflovs/supervs-install-english.pdf#page=9) for Superflo VST it states:  

> **Note:** IntelliConnect Control and Monitoring Systems can NOT externally control this pump via RS-485. The pump will need to be connected via digital inputs.

It's unclear why this restriction is in place, possibly the newer Superflo/Whisperflo VST pumps weren't tested against the IntelliConnect.  But if there is technical limitation, it seems conceivable it could be resolved with a firmware update.

**Reply by rstrouse** [reply]
Do you by chance have an IntelliConnect lying around?

**Reply by mguinness** [reply]
I don't unfortunately.  It's more of a curiosity on my part why these new pumps wouldn't be compatible with that system.

**Reply by rstrouse** [reply]
I think if you read the Pentair Home manual you will see what it is doing.  The IntelliFlo series pumps are using the mode and schedule since there is no scheduling on the controller itself.  It is relying on the 4 schedules that are stored on the pump.   Booh to that!

This means that the first 4 modes (speeds) and the second 4 modes (schedules) are also stored on the pump.  This scheduling function does not exist on the drives attached to the SuperFlo/WhisperFlo pumps but I am not sure that they don't still have 8 modes to represent speeds.   Maybe there are only the 4 speed modes.

IntelliConnect is simply the old IntelliSync controller that has had some updates and all scheduling appears to be done on the pump.  The only thing that I can figure is that they dusted this off since they were having trouble with supply.  It is a considerable step back for their automation platforms.

That being said EasyTouch, IntelliTouch, IntelliCenter, and Nixie do not use the scheduler on the pump and thus would not need that protocol available.  I can imagine that this would be a limitation for the old IntelliFlo 4x160 pumps as well since while it has 4 scheduler modes the time is updated externally.

**Reply by mguinness** [reply]
Great detective work!  The Pentair Home for [IntelliConnect](https://www.pentair.com/content/dam/extranet/nam/pentair-pool/residential/automation/intelliconnect/manuals---user-documentation/users-guide/intelliconnect-link20-app-guide.pdf#page=9) manual has the following which I guess tipped you off.

> Programs 1-4 of an IntelliFlo pump can be controlled and programmed from the IntelliFlo screen

So the RS-485 connection is used for setting the speeds and schedule on the Intelliflo from the IntelliConnect.  The pump status can also read by IntelliConnect.

![ic_app](https://user-images.githubusercontent.com/14002844/185977088-67c5b55c-35d3-46f8-917e-6ea7cd0b34fd.png)

The Pentair Home for [IntelliSync](https://www.pentair.com/content/dam/extranet/nam/pentair-pool/residential/automation/intellisync/manuals/intellisync-pentair-home-users-guide.pdf#page=12) manual has the following under supported devices which does list Superflo VST.

> Pumps: IntelliFlo (IntelliPro), SuperFlo VST (SuperMax VS), or WhisperFloXF VS (Max-E-ProXF VS)

However the Superflo only has 1 scheduled start time with up to 3 different speeds of varying duration running consecutively.

**Comment by mguinness** [comment]
Not wanting to hijack the post, but I wanted to share the Century VGreen [Gen3 EPC ModBus Comm Protocol](https://www.troublefreepool.com/attachments/gen3-epc-modbus-communication-protocol-_rev4-17-pdf.358807/) document.

Unsure if this would allow control from Nixie, but it appears to have some compatibility for Jandy/Pentair/Hayward protocols. 

![regal](https://user-images.githubusercontent.com/14002844/186285638-196a88c4-b843-4856-bad7-5d1fce86c0ab.png)

**Reply by zombielinux** [reply]
Replying that I just tried connecting it with the Pentair option enabled. I'm not able to communicate with it using the Intelliflo pump type on any address. 

Jandy support would be terrific, because I know that works.

**Reply by mguinness** [reply]
Looks like Century (Regal) Modbus pump support was added in https://github.com/tagyoureit/nodejs-poolController/pull/1120

**Comment by mguinness** [comment]
FWIW, the Nidec Neptune motor has support for [External Pool Control System](https://acim.nidec.com/motors/usmotors/-/media/USMotors/Documents/TechDocs/Manuals/NEPTUNE-Variable-Speed-Pool-Pump-Motor-Manual--150THP-165THP-225THP-27THP-and-345THP.pdf#page=43) via RS-485.

|Manufacturer|Control System|
|--- |--- |
|Hayward®|OmniLogic®, Omni® Smart, OmniHub®, OmniPL®|
|Jandy®/Fluidra®|Aqualink® TCX, iAqualink® 3.1, Aqualink® RS, JEP-R®|
|Pentair®|EasyTouch® PL4 & PSL4, Intelliconnect®, IntelliCenter®, IntelliCenter® Lite|
|Waterway™|OASIS|
|PoolSide Tech|The Attendant|

DIP switch settings for each control system as follows.

|Control System|Switch1|Switch 2|Switch 3|Switch 4|
|--- |--- |--- |--- |--- |
|Hayward®|OFF|ON|OFF|OFF|
|Jandy® / Fluidra®|ON|OFF|OFF|OFF|
|Pentair®|OFF|ON|ON|OFF|
|Waterway™|OFF|OFF|ON|OFF|
|PoolSide Tech|OFF|OFF|OFF|OFF|

It also appears that the same [Drive Interface Board](https://waterwayplastics.com/manuals/M018324880000PowerDefender165-270forWaterway.pdf#page=27) is used in the Waterway Power Defender pumps.

Support for control via Modbus has been added to nodejs poolController in commit https://github.com/tagyoureit/nodejs-poolController/commit/b814bb3.
